### ISSUE TITLE
Create A Generic Test Suite For All Path Planners

### ISSUE THREAD

**Issue by garethellis0** [UBC-Thunderbots/Software#770]
### Description of the task
Create a generic suite of tests to make sure we can verify that all path planners obey sane behavior. This will likely take the form of designing a series of scenarios that we expect all path planners to be able to handle and writing tests for them. This will allow us to more rapidly bring up new path planner algorithms in the future with much greater confidence.

### Acceptance criteria

- [ ] Implement a series of unit tests that represent the set of scenarios we expect all path planners to be able to handle.
- [ ] Test all path planners with the test suite, fix (or file tickets to fix) any bugs found

### Blocked By

#769 (and any tickets created from it)

**Comment by MathewMacDougall** [comment]
@jordanzliu @garethellis0 @jonathanlew 

Jordan is going to start work on this, but I want to clarify some stuff first since we made this ticket before the navigator re-architecture.

The just want to clarify this will be a generic test suite for `Path Planner` (ie. the part that specifically navigates from point A to B), not any higher-level constructs like `PathManagers`

**Comment by jonathanlew** [comment]
> 
> 
> @jordanzliu @garethellis0 @jonathanlew
> 
> Jordan is going to start work on this, but I want to clarify some stuff first since we made this ticket before the navigator re-architecture.
> 
> The just want to clarify this will be a generic test suite for `Path Planner` (ie. the part that specifically navigates from point A to B), not any higher-level constructs like `PathManagers`

Just testing `Path Planner` sounds good to me. Although we'll eventually want a `PathManager` test, so maybe those can be co-developed or we can make a new issue.

**Comment by garethellis0** [comment]
Yeah, I agree. We should work bottom up, and the path planner is the first level from the bottom that does not have a comprehensive set of tests.
